src/TiledArray/conversions/foreach.h

```diff
@@ -265,15 +487,16 @@ namespace TiledArray {
   /// of a tile is held in a \c std::shared_ptr. If you need to ensure other
   /// copies of the data are not modified or this behavior causes problems in
   /// your application, use the \c TiledArray::foreach function instead.
-  template <typename Tile, typename Op>
+  template <typename Tile, typename Op,
+      typename = typename std::enable_if<! TiledArray::detail::is_array<typename std::decay<Op>::type>::value>::type>
```

Comment:
This is necessary because `op` is a universal reference and will be selected as the "best" match versus other overloaded functions (e.g. line 632).